tests/phpunit/includes/Deserializers/ExpDataDeserializerTest.php

```diff
@@ -0,0 +1,154 @@
+<?php
+
+namespace SMW\Tests\Deserializers;
+
+use SMW\Deserializers\ExpDataDeserializer;
+use SMW\Serializers\ExpDataSerializer;
+use SMW\Exporter\Element\ExpResource;
+use SMW\Exporter\Element\ExpNsResource;
+use SMW\Exporter\Element\ExpLiteral;
+use SMWExpData as ExpData;
+use SMWDIBlob as DIBlob;
+
+/**
+ * @covers \SMW\Deserializers\ExpDataDeserializer
+ *
+ * @group semantic-mediawiki
+ *
+ * @license GNU GPL v2+
+ * @since 2.2
+ *
+ * @author mwjames
+ */
+class ExpDataDeserializerTest extends \PHPUnit_Framework_TestCase {
+
+	public function testCanConstructor() {
+
+		$this->assertInstanceOf(
+			'\SMW\Deserializers\ExpDataDeserializer',
+			new ExpDataDeserializer()
+		);
+	}
+
+	public function testInvalidSerializerObjectThrowsException() {
+
+		$instance = new ExpDataDeserializer();
+
+		$this->setExpectedException( 'OutOfBoundsException' );
+		$instance->deserialize( 'Foo' );
+	}
+
+	public function testVersionMismatchThrowsException() {
+
+		$instance = new ExpDataDeserializer();
+
+		$this->setExpectedException( 'OutOfBoundsException' );
+		$instance->deserialize( array( 'version' => 0.2 ) );
+	}
+
+	/**
+	 * @dataProvider expDataProvider
+	 */
+	public function testDeserialize( $seralization, $expected ) {
+
+		$instance = new ExpDataDeserializer();
+
+		$this->assertEquals(
+			$expected,
+			$instance->deserialize( $seralization )
+		);
+
+		$this->assertEquals(
+			$expected->getHash(),
+			$instance->deserialize( $seralization )->getHash()
+		);
+	}
+
+	public function expDataProvider() {
```

Comment:
I highly encourage you to create dedicated test methods for different cases being tested. That makes it a lot easier to understand what is being tested.